modules/cross_platform.py

```diff
@@ -1,11 +1,42 @@
 def fix_path(path, platform):
-    if platform == "windows":
-        path.replace("/", "\\")
+    if platform == 'windows':
+        path.replace('/', '\\')
     else:
-        path.replace("\\", "/")
+        path.replace('\\', '/')
     return path
 
 def chdir(directory, platform):
     import os
     if os.path.basename(os.getcwd()) != directory:
         os.chdir(fix_path(directory, platform))
+
+def tabprint(array):
+    result = ''
+    length = 0
+    pos = 0
+    # Get length of longest string + 5
+    for item in array:
+        if len(item) > length:
+            length = len(item) + 5
+    # Add items to string to print
+    for item in array:
+        spacer = ''
+        for i in range(length - len(item)):
+            spacer += ' '
+        pos += len(item) + len(spacer)
+        if pos <= term_width() - length:
+            result += item + spacer
+        else:
+            pos = 0
+            result += '\n'
+    print(result)
+
+def term_width():
+    import os
+    rows, columns = os.popen('stty size', 'r').read().split()
```

Comment:
what does this instruction do?

Reply:
This returns the terminal width for printing tables. The terminal width is used so that row length can be calculated and spacing used. The command 'stty size' returns 2 numbers with a space in between. ex.('24 80' for a 24x80 terminal window.) The .read().split() assigning two variables for the rows and columns length. I return the one I want for that function.